In [7]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

Q1)

We are going to conduct a 2 tailed t-Test on 2 Independent samples with Numerical Data.

We need to check whether the mean of both samples are different and is there any significance difference between the two samples?


Null Hypothesis is want we want to prove

Null Hypothesis: μ1=μ2

Alternative Hypthosis: μ1 NOT EQUAL TO μ2






Significance = 5%

alpha = 0.05

As it is a two-tailed test

alpha/2 = 0.025


In [8]:
# Loading dataset
cutlets=pd.read_csv('Cutlets.csv')
cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [9]:
cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [10]:
unitA=pd.Series(cutlets.iloc[:,0])
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [11]:
unitB=pd.Series(cutlets.iloc[:,1])
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [12]:
# 2-sample 2-tail ttest:   stats.ttest_ind(array1,array2)     # ind -> independent samples
p_value=stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

Compare pvalue with α = 0.05 (At 5% significance level)

As pvalue is 0.472 which is not equal to α=0.5

Hence, We fail to reject Null Hypothesis because of lack of evidence, there is no significant difference between the two samples.


Q2)

We are going to conduct a ANOVA Test on 4 Independent samples with Numerical Data

We need to check whether the mean of any of these samples are different or the same?

Null Hypothesis is want we want to prove

Null Hypothesis: μ1=μ2=μ3=μ4

Alternative Hypthosis: Atleast One of them is Differente

Significance 5%

alpha = 0.05

In [13]:
# Loading dataset
labtat=pd.read_csv('LabTAT.csv')
labtat.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [14]:
labtat.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [15]:
#lab1=pd.Series(labtat.iloc[:,0]) 
#lab1

In [16]:
#lab2=pd.Series(labtat.iloc[:,1]) 
#lab2

In [17]:
# Anova ftest statistics: stats.f_oneway(column-1,column-2,column-3,column-4)
pvalue=stats.f_oneway(labtat.iloc[:,0],labtat.iloc[:,1],labtat.iloc[:,2],labtat.iloc[:,3])
pvalue

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Compare pvalue with α = 0.05 (At 5% significance level)

As pvalue is 2.115 which is not equal to α=0.5

Hence, We fail to reject Null Hypothesis because of lack of evidence, there is no significant difference between the two samples.

Q3)

We are going to conduct a Test of Independence using Chi-Square test with Contingency table.

We need to check whether the proportion of any of these samples are different or the same?

Null Hypothesis is want we want to prove

Null Hypothesis: There is no association or dependency between the gender based buyer rations across regions

Alternative Hypthosis: There is a significant association or dependency between the gender based buyer rations across regions


Significance 5%

alpha = 0.05

As it is a one-tailed test

alpha = 1-0.95 = 0.05


In [18]:
# Loading dataset
buyerratio=pd.read_csv('BuyerRatio.csv')
buyerratio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [19]:
buyerratio_table=buyerratio.iloc[:,1:6]
buyerratio_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [20]:
buyerratio_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [21]:
val=stats.chi2_contingency(buyerratio_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [22]:
observed = np.array([50, 142, 131, 70, 435, 1523, 1356, 750])
expected = np.array([42.76531299,  146.81287862,  131.11756787, 72.30424052, 442.23468701, 1518.18712138, 1355.88243213, 747.69575948])

In [23]:
#Comparing Evidence with Hypothesis
statistics, p_value = stats.chisquare(observed, expected, ddof = 3)
print("Statistics = ",statistics,"\n",'P_Value = ', p_value)

Statistics =  1.5959455390914483 
 P_Value =  0.8095206646905712


Compare pvalue with α = 0.05 (At 5% significance level)

As pvalue is 0.809 which is not equal to α=0.5

We fail to reject Null Hypothesis because of lack evidence. Therefore, there is no association or dependency between male-female buyers rations and are similar across regions. Hence, Independent samples.

4Q)

We are going to conduct a Test of Independence using Chi-Square test with Contingency table.

We need to check whether the mean of any of these samples are different or the same?

Null Hypothesis is want we want to prove

Null Hypothesis: μ1=μ2=μ3=μ4

Alternative Hypthosis: Atleast One of them is Differente

Significance 5%

alpha = 0.05

In [24]:
# Loading dataset
cof=pd.read_csv('Costomer+OrderForm.csv')
cof

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [25]:
cof.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [26]:
#Checking value counts in data
print(cof['Phillippines'].value_counts(),'\n',cof['Indonesia'].value_counts(),'\n',cof['Malta'].value_counts(),'\n',cof['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 
 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 
 Error Free    269
Defective      31
Name: Malta, dtype: int64 
 Error Free    280
Defective      20
Name: India, dtype: int64


In [27]:
#Creating Contingency table
contingency_table = [[271,267,269,280],
                    [29,33,31,20]]
print(contingency_table)

[[271, 267, 269, 280], [29, 33, 31, 20]]


In [28]:
#Calculating Expected Values for Observed data
stat, p, df, exp = stats.chi2_contingency(contingency_table)
print("Statistics = ",stat,"\n",'P_Value = ', p,'\n', 'degree of freedom =', df,'\n', 'Expected Values = ', exp)

Statistics =  3.858960685820355 
 P_Value =  0.2771020991233135 
 degree of freedom = 3 
 Expected Values =  [[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [29]:
#Defining Expected values and observed values
observed = np.array([271, 267, 269, 280, 29, 33, 31, 20])
expected = np.array([271.75, 271.75, 271.75, 271.75, 28.25, 28.25, 28.25, 28.25])

In [30]:
#Compare Evidences with Hypothesis using t-statictic
test_statistic , p_value = stats.chisquare(observed, expected, ddof = df)
print("Test Statistic = ",test_statistic,'\n', 'p_value =',p_value)

Test Statistic =  3.858960685820355 
 p_value = 0.4254298144535761


Compare pvalue with α = 0.05 (At 5% significance level)

As pvalue is 0.425 which is not equal to α=0.5

We fail to reject Null Hypothesis because of lack of evidence.